### PR TITLE
docs(demos): add a variety of button demos

### PR DESCRIPTION
**Pull request by amcdnl**
Adds a variety of new button demos and combinations.

buttons

### CODE REVIEW

**Comment by jelbourn** [comment]
For the user-facing demo, the fabs should have icons inside rather than text

File: src/material-examples/button-types/button-types-example.html [hunk 1]
```diff
@@ -1,17 +1,48 @@
...
-    <button md-button>Flat button</button>
-    <button md-raised-button>Raised button</button>
-    <button md-fab><md-icon>check</md-icon></button>
-    <button md-mini-fab><md-icon>check</md-icon></button>
-  </div>
+<h3>Basic Buttons</h3>
+<div class="button-row">
+  <button md-button>Basic</button>
+  <button md-button color="primary">Primary</button>
+  <button md-button color="accent">Ascent</button>
```
Comment (jelbourn) [inline src/material-examples/button-types/button-types-example.html]: "Ascent" -> "Accent" (throughout file)

File: src/material-examples/button-types/button-types-example.html [hunk 2]
```diff
@@ -1,17 +1,48 @@
...
+  <a md-raised-button routerLink=".">Link</a>
+</div>
+
+<h3>Icon Buttons</h3>
+<div class="button-row">
+  <button md-icon-button><md-icon class="md-24">favorite</md-icon></button>
+  <button md-icon-button color="primary"><md-icon class="md-24">favorite</md-icon></button>
+  <button md-icon-button color="accent"><md-icon class="md-24">favorite</md-icon></button>
+  <button md-icon-button color="warn"><md-icon class="md-24">favorite</md-icon></button>
+  <button md-icon-button disabled><md-icon class="md-24">favorite</md-icon></button>
```
Comment (jelbourn) [inline src/material-examples/button-types/button-types-example.html]: Each icon button needs an `aria-label`

File: src/material-examples/button-types/button-types-example.html (hunk 1 above)
Comment (amcdnl) [inline src/material-examples/button-types/button-types-example.html]: LOL!

**Comment by amcdnl** [comment]
@jelbourn - ready for further review.

File: src/material-examples/button-types/button-types-example.html [hunk 3]
```diff
@@ -1,17 +1,50 @@
...
+  <button md-raised-button color="accent">Accent</button>
+  <button md-raised-button color="warn">Warn</button>
+  <button md-raised-button disabled>Disabled</button>
+  <a md-raised-button routerLink=".">Link</a>
+</div>
+
+<h3>Icon Buttons</h3>
+<div class="button-row">
+  <button md-icon-button><md-icon class="md-24" aria-label="favorite">favorite</md-icon></button>
+  <button md-icon-button color="primary"><md-icon class="md-24" aria-label="favorite">favorite</md-icon></button>
```
Comment (jelbourn) [inline src/material-examples/button-types/button-types-example.html]: Should be a bit more specific w/ the aria-label:
`"Example icon-button with a heart icon"`

**Pull request merged**